src/components/switch/switch.tsx

```diff
@@ -86,28 +88,44 @@ export class Switch {
 
     public render() {
         return [
-            <div
+            <button
+                id={this.fieldId}
                 class={{
                     'mdc-switch': true,
-                    'mdc-switch--disabled': this.disabled || this.readonly,
                     'lime-switch--readonly': this.readonly,
+                    'mdc-switch--unselected': !this.value,
+                    'mdc-switch--selected': this.value,
                 }}
+                type="button"
+                role="switch"
+                aria-checked={this.value}
+                disabled={this.disabled || this.readonly}
+                onClick={this.handleClick}
             >
                 <div class="mdc-switch__track" />
-                <div class="mdc-switch__thumb-underlay">
-                    <div class="mdc-switch__thumb">
-                        <input
-                            type="checkbox"
-                            class="mdc-switch__native-control"
-                            id={this.fieldId}
-                            role="switch"
-                            onChange={this.onChange}
-                            disabled={this.disabled || this.readonly}
-                            checked={this.value}
-                        />
+                <div class="mdc-switch__handle-track">
+                    <div class="mdc-switch__handle">
+                        <div class="mdc-switch__shadow">
+                            <div class="mdc-elevation-overlay"></div>
+                        </div>
+                        <div class="mdc-switch__ripple"></div>
+                        <div class="mdc-switch__icons">
+                            <svg
+                                class="mdc-switch__icon mdc-switch__icon--on"
+                                viewBox="0 0 24 24"
+                            >
+                                <path d="M19.69,5.23L8.96,15.96l-4.23-4.23L2.96,13.5l6,6L21.46,7L19.69,5.23z" />
+                            </svg>
+                            <svg
+                                class="mdc-switch__icon mdc-switch__icon--off"
+                                viewBox="0 0 24 24"
+                            >
+                                <path d="M20 13H4v-2h16v2z" />
+                            </svg>
```

Comment:
![image](https://user-images.githubusercontent.com/35954987/153393849-8e353710-5dbb-46f5-bef2-a9b5e01a23d1.png)

Even though I understand where they come from, I dislike the visual mess these icons create. I think the direction and color are enough for users to realize that they are ON or OFF. 

- We don't care that much about RTL layout neither, do we? 
- And I don't think we want anyone to use a grey color as their accent color (to make the switch confusing), right?

Reply:
I'd guess the icons are a matter of being colour-blind friendly?

I actually kind of like the icons. Since we're using green as the "on" colour, and I am fully capable of distinguishing green from grey, I find our switches _fairly_ clear anyway. At least when _on_. When off, if there are no "on" switches to compare with, the state is probably less obvious. But I've seen switches that are light grey (dull) when "off", and white (bright and shiny) when "on". I think my Android phone has that colour scheme. And even though I know "left" typically means "off", and "right" typically means "on", _every single bloody time_ I come across one of those switches, I have to flip it once to check if it looks "more on" or "more off" in the other position, just to be sure 🤮

And while it's easy to say "those switches are obviously using a much worse colour scheme than ours are", keep in mind that _someone_ must have thought those switches are clear enough, just like we think _our_ switches are clear enough. We see our switches every day, and know by heart what the "opposite" state looks like. While we can make an effort to imagine our components from the view of a first time user of them, it's impossible for any of us to truly rid ourselves of that contextual knowledge.

Reply:
> I have to flip it once to check if it looks "more on" or "more off" in the other position, just to be sure

I also do this a lot :smile:

Reply:
> This opinion probably betrays my age, right?

😸  certainly not. I think you're young enough to have experienced both. 

But it's super interesting that you find this toggle switch confusing. 🤔  I really can't explain why, but I haven't used android for at least 5 years now. I just know that I always disliked details of material design, in comparison with iOS. I also always strongly disliked their switches and can't really say why. (maybe that's why I tried to make the switch slightly different this time). 

But this switch you showed is more like the iOS style. The only problem I'd say is that the thumb is greyed-out which to me looks **disabled-and-off**. 

Maybe that's why it makes you confused too? 🤔 

I guess to be able to explain why you are confused by the switch, we need to check how the switches look like and behave across their platform and apps. Probably there is an inconsistency there that creates this issue.

Reply:
I remember Android settings has checkboxes, radios and switches all over the place and at leas the time I worked with that at Sony Mobile, it was a very confusing and inconsistent thing. There was an Android guideline when to use checkboxes vs. switches though. They said whenever there is a SAVE button, you can use a checkbox, and if the thing auto-applies or auto-saves, use switches. But do you think anyone in Sony Mobile (who customized the hell of Android, and had their own Sony apps) knew or followed that guideline?

iOS settings on the other hand don't have these variety of things. It's mostly switches and a very specific way of having radios (which don't look like radios 😄 ). 👇 

![image](https://user-images.githubusercontent.com/35954987/153569801-bd589e68-9552-4a1e-a3a1-41c411eb63df.png)

Anyhow...

The only legitimate explanation for me is that switches probably confuse the hell out of RTL users, since some of the apps don't care to adjust their UI elements and some do and the direction in which the switch is ON or OFF (along with those inconsistencies) make them very confused. So a checkmark is probably a good addition.